src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer_ReleaseTracking.cs

```diff
@@ -0,0 +1,699 @@
+            helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md",
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        internal static readonly DiagnosticDescriptor InvalidRemovedWithoutShippedEntryInAnalyzerReleasesFileRule = new DiagnosticDescriptor(
+            id: DiagnosticIds.InvalidEntryInAnalyzerReleasesFileRuleId,
+            title: CodeAnalysisDiagnosticsResources.InvalidEntryInAnalyzerReleasesFileRuleTitle,
+            messageFormat: CodeAnalysisDiagnosticsResources.InvalidRemovedWithoutShippedEntryInAnalyzerReleasesFileRuleMessage,
+            category: DiagnosticCategory.MicrosoftCodeAnalysisReleaseTracking,
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: CodeAnalysisDiagnosticsResources.InvalidEntryInAnalyzerReleasesFileRuleDescription,
+            helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md",
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private static bool TryGetReleaseTrackingData(
+            ImmutableArray<AdditionalText> additionalTexts,
+            CancellationToken cancellationToken,
+            [NotNullWhen(returnValue: true)] out ReleaseTrackingData? shippedData,
+            [NotNullWhen(returnValue: true)] out ReleaseTrackingData? unshippedData,
+            out List<Diagnostic>? invalidFileDiagnostics)
+        {
+            if (!TryGetReleaseTrackingFiles(additionalTexts, cancellationToken, out var shippedText, out var unshippedText))
+            {
+                // TODO: Report a diagnostic that both must be specified if either shippedText or unshippedText is non-null.
+                shippedData = default;
+                unshippedData = default;
+                invalidFileDiagnostics = null;
+                return false;
+            }
+
+            invalidFileDiagnostics = new List<Diagnostic>();
+            shippedData = ReadReleaseTrackingData(shippedText.Path, shippedText.GetText(cancellationToken), invalidFileDiagnostics.Add, isShippedFile: true);
+            unshippedData = ReadReleaseTrackingData(unshippedText.Path, unshippedText.GetText(cancellationToken), invalidFileDiagnostics.Add, isShippedFile: false);
+
+            return invalidFileDiagnostics.Count == 0;
+        }
+
+        private static bool TryGetReleaseTrackingFiles(
+            ImmutableArray<AdditionalText> additionalTexts,
+            CancellationToken cancellationToken,
+            [NotNullWhen(returnValue: true)] out AdditionalText? shippedText,
+            [NotNullWhen(returnValue: true)] out AdditionalText? unshippedText)
+        {
+            shippedText = null;
+            unshippedText = null;
+
+            StringComparer comparer = StringComparer.Ordinal;
```

Comment:
>Ordinal [](start = 53, length = 7)

OrdinalIgnoreCase? Or try Ordinal first, fallback to OrdinalIgnoreCase?

Reply:
Changed to `OrdinalIgnoreCase`, thanks.